numpy/core/_asarray.py

```diff
@@ -23,9 +23,12 @@ def asarray(a, dtype=None, order=None):
         of lists and ndarrays.
     dtype : data-type, optional
         By default, the data-type is inferred from the input data.
-    order : {'C', 'F'}, optional
-        Whether to use row-major (C-style) or
-        column-major (Fortran-style) memory representation.
+    order : {'K', 'A', 'C', 'F'}, optional
+        Memory layout.  'K' and 'A' depend on the order of input array a: 
+        'K' preserve input order
+        'A' preserve input if 'F', otherwise convert to 'C',
```

Comment:
nit: We can make this similar to np.copy to make it more clearer.

```suggestion
        'A' means 'F' if `a` is Fortran contiguous, 'C' otherwise.
```